homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -224,27 +348,37 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
         host = config_entry.data[CONF_HOST]
         token = config_entry.data[CONF_TOKEN]
         model = config_entry.data[CONF_MODEL]
-        device = None
+        device = hass.data[DOMAIN][config_entry.entry_id].get(KEY_DEVICE)
         sensors = []
         if model in (MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1):
-            device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
             sensors = HUMIDIFIER_CA1_CB1_SENSORS
         elif model in MODELS_HUMIDIFIER_MIOT:
-            device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
             sensors = HUMIDIFIER_MIOT_SENSORS
         elif model in MODELS_HUMIDIFIER_MJJSQ:
-            device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
             sensors = HUMIDIFIER_MJJSQ_SENSORS
         elif model in MODELS_HUMIDIFIER_MIIO:
-            device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
             sensors = HUMIDIFIER_MIIO_SENSORS
+        elif model == MODEL_AIRPURIFIER_V2:
```

Comment:
Perhaps we could do the model to sensors translation using a constant dict

Reply:
Good idea.